source/modules/sensors/SensorsGetters.mc

```diff
@@ -268,16 +268,24 @@ module SensorsGetters {
             return System.getDeviceSettings().phoneConnected;
         }
 
-        function getHR() as Number? {
+        function getHR() as Array<Number?>? {
             var activityInfo = Activity.getActivityInfo();
             var value = activityInfo != null ? activityInfo.currentHeartRate : null;
-
+            var isHeartWorking = true;
             if (value == null && Checkers.checkHeartRateHistory()) {
                 var sensorInfo = SensorHistory.getHeartRateHistory({}).next();
                 value = sensorInfo != null ? sensorInfo.data : value;
+                if (value != null) {
+                    isHeartWorking = false;
+                }
             }
-
-            return value != null && value < ActivityMonitor.INVALID_HR_SAMPLE ? value.toNumber() : null;
+            if (value == null) {
+                isHeartWorking = false;
+            }
+            return [
+                value != null && value < ActivityMonitor.INVALID_HR_SAMPLE ? value.toNumber() : null,
+                isHeartWorking == true ? 0 : 1
+            ];
```

Comment:
```suggestion
            return [
                value != null && value < ActivityMonitor.INVALID_HR_SAMPLE ? value.toNumber() : null,
                value == null ? 0 : 1
            ];
```
We can simplify this code by removing unnecessary variables. I also suggest using the numbers 0 and 1 for the icon as an indicator that the data is either not present or is present. When HR is not displaying information it is 0 when there is data it is 1, this way it will be easier to follow the logic and understand which icon is needed for what purpose

Reply:
OK,Thanks for the correction.